src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8879,6 +8892,11 @@ private ArgumentSyntax ParseArgumentExpression(bool isIndexer)
                 expression = this.ParseSubExpression(0);
             }
 
+            if (refOrOutKeyword != null && IsLambda(expression.Kind))
+            {
+                expression = this.AddError(expression, ErrorCode.ERR_RefLvalueExpected);
```

Comment:
I am not suggesting to report a different error, just saying that the parser is probably the wrong place to report it.